core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/util/ClassLoaderHelper.java

```diff
@@ -25,22 +25,20 @@ private ClassLoaderHelper() {
 
     private static Class<?> loadClassViaClasses(String fqcn, Class<?>[] classes) {
         if (classes != null) {
-            for (Class<?> c : classes) {
```

Comment:
The behavior seems different. In old code, we would use class loader of each class if loading class doesn't work. But in new code, we only use the loader for the first class that returns non-null getClassLoader.

Reply:
Oops.. Good catch, will fix!